backend/src/main/controller/PromotionController.ts

```diff
@@ -0,0 +1,188 @@
+import { NextFunction, Request, Response } from 'express';
+import { PromotionRepository } from '../repository/PromotionRepository';
+import { getCustomRepository, SelectQueryBuilder } from 'typeorm';
+import { Promotion } from '../entity/Promotion';
+import { UserRepository } from '../repository/UserRepository';
+import { Discount } from '../entity/Discount';
+import {
+  PromotionDTO,
+  PromotionValidation,
+} from '../validation/PromotionValidation';
+import {
+  PromotionQueryDTO,
+  PromotionQueryValidation,
+} from '../validation/PromotionQueryValidation';
+import { IdValidation } from '../validation/IdValidation';
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+export class PromotionController {
+  /**
+   * Retrieves all promotions and their discounts
+   * * First we need to validate the query params and cast that into a PromotionQueryDTO
+   * * Then we apply the query options into the query builder depending on which properties are present
+   * * Now we execute the query builder and return its results
+   */
+  getAllPromotions = async (
+    request: Request,
+    response: Response,
+    next: NextFunction
+  ): Promise<any> => {
+    try {
+      let queryBuilder = getCustomRepository(PromotionRepository)
+        .createQueryBuilder('promotion')
+        .innerJoinAndSelect('promotion.discount', 'discount');
+
+      if (request.query) {
+        const promotionQuery: PromotionQueryDTO = await PromotionQueryValidation.schema.validateAsync(
+          request.query,
+          { abortEarly: false }
```

Comment:
`abortEarly: false` basically tells Joi to continue validating even if there is already an error. This way it'll give us back ALL the validation issues at once, so if multiple parts of the request are ill-formatted itll tell us all parts at once.

For example, if the incoming cuisine type isn't a valid `CuisiineType` and the expiration_date isn't a valid date, it'll give us back both validation errors rather than only the first one it encounters